cpp/src/parquet/arrow/writer.cc

```diff
@@ -333,7 +340,7 @@ class FileWriterImpl : public FileWriter {
           std::unique_ptr<ArrowColumnWriterV2> writer,
           ArrowColumnWriterV2::Make(*data, offset, size, schema_manifest_,
                                     row_group_writer_));
-      return writer->Write(&column_write_context_);
+      return writer->Write(&column_write_context_.back());
```

Comment:
Would a writer doesn't have any columns?

Reply:
Then it is invalid to call `WriteColumnChunk`. Maybe I need to add a check to protect it.